snuba/query/mql/parser.py

```diff
@@ -0,0 +1,636 @@
+from __future__ import annotations
+
+import logging
+from typing import Any, Callable, Mapping, Optional, Sequence, Tuple, Union
+
+import sentry_sdk
+from parsimonious.nodes import Node, NodeVisitor
+from snuba_sdk.conditions import OPERATOR_TO_FUNCTION, ConditionFunction, Op
+from snuba_sdk.dsl.dsl import MQL_GRAMMAR
+
+from snuba.datasets.dataset import Dataset
+from snuba.datasets.entities.entity_key import EntityKey
+from snuba.datasets.entities.factory import get_entity
+from snuba.query import OrderBy, OrderByDirection, SelectedExpression
+from snuba.query.composite import CompositeQuery
+from snuba.query.conditions import binary_condition, combine_and_conditions
+from snuba.query.data_source.simple import Entity as QueryEntity
+from snuba.query.exceptions import InvalidQueryException
+from snuba.query.expressions import Column, FunctionCall, Literal
+from snuba.query.logical import Query as LogicalQuery
+from snuba.query.parser.exceptions import ParsingException
+from snuba.query.query_settings import QuerySettings
+from snuba.query.snql.anonymize import format_snql_anonymized
+from snuba.query.snql.parser import (
+    POST_PROCESSORS,
+    VALIDATORS,
+    _post_process,
+    _replace_time_condition,
+    _treeify_or_and_conditions,
+)
+from snuba.state import explain_meta
+
+logger = logging.getLogger("snuba.mql.parser")
+
+
+class MQLVisitor(NodeVisitor):
```

Comment:
The visitor on the SDK transforms the MQL into a `MetricsQuery`, not a snuba AST.

Reply:
I understand that part but what I'm wondering is, can we transform the `MetricsQuery` to a snuba AST?

Reply:
We wouldn't need to! The new endpoint will receive a MQL string. To reduce the overhead of query transformations we do (which is already a lot), we could directly go from MQL string->  AST rather than MQL string -> MetricQuery -> AST. Does that make sense?

Reply:
Will the SDK still need to transform into a MetricQuery?

Reply:
In a world, where the new endpoint is fully functional and product folks fully adopt MQL, no. But for now, we still need MetricsQuery.